### PR TITLE
Development-restricted mergeback

### PR DESCRIPTION
**Pull request by mpg**
## Description

This brings into development:
- the security fixes that were part of the 3.6.1 release;
- some bignum constant-time improvements.

## PR checklist

- [ ] **changelog** not required because: mergeback, ChangeLog entries already included in individual PRs
- [ ] **development PR** here
- [ ] **framework PR** not required
- [ ] **3.6 PR** not required because: changes already in 3.6.1
- [ ] **2.28 PR** not required because: applicable changes already in 2.28.9
- **tests**  not required because: mergback

### CODE REVIEW

**Review by yanesca — APPROVED** [review]
Performed the merge locally and arrived at the same result. LGTM.

**Review by gilles-peskine-arm — APPROVED** [review]
This is a correct merge of `development-restricted` into `development`.

Maybe now would be a good time to run `assemble_changelog`, to mark a milestone? Or should we wait until we merge the couple of straggler forward-ports?

**Comment by mpg** [comment]
> Maybe now would be a good time to run assemble_changelog, to mark a milestone? Or should we wait until we merge the couple of straggler forward-ports?

I'm not sure if it's useful, but I'd say if we do it, I'd rather wait for the forward-ports.

**Pull request merged**
